packages/document-drive/src/queue/base.ts

```diff
@@ -53,38 +59,29 @@ export class MemoryQueue<T> implements IQueue<T> {
     return this.id;
   }
 
-  async setBlocked(blocked: boolean) {
-    this.blocked = blocked;
-  }
-
   async isBlocked() {
-    return this.blocked;
+    return this.running || this.deleted || this.dependencies.length > 0;
```

Comment:
The switch to `running` instead of a `blocked` seems less fault tolerant (eg - any code path that doesn't set it back to false will block indefinitely), and particularly because anything can toggle it externally. Why does that need to be public?

Reply:
Maybe an enum would be clearer? `Idle`, `Running`, `Deleted`, etc?